typed-racket-lib/typed-racket/base-env/base-env-numeric.rkt

```diff
@@ -70,14 +70,14 @@
     (-> t1 t2 B))
   ;; simple case useful with equality predicates.
   ;; if the equality is true, we know that general arg is in fact of specific type.
-  (define (commutative-equality/filter general specific)
-    (list (-> general specific B : (-FS (-filter specific 0) -top))
-          (-> specific general B : (-FS (-filter specific 1) -top))))
+  (define (commutative-equality/prop general specific)
+    (list (-> general specific B : (-PS (-is-type 0 specific) -tt))
```

Comment:
It read left-to-right more clearly to me as (-is-type x t), since it then reads in the same order as "x is of type t"